### PR TITLE
Implements SpiralGroundCurve::GInverse().

### PR DESCRIPTION
**Pull request by agalbachicar**
# 🎉 New feature

Part of #265
Goes on top of #270

## Summary

Implements `GInverse()`. Tested against the normalized spiral. With htis PR the curve implementation is complete and validated against the normalized spiral. Future PRs will work on non-normalized spirals.

## Test it

Via unit tests.

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if it affects the public API)

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Review by stonier — APPROVED** [review]
LGTM Pending Franco's review.

File: src/maliput_malidrive/road_curve/spiral_ground_curve.cc [hunk 1]
```diff
@@ -184,9 +184,89 @@ maliput::math::Vector2 SpiralGroundCurve::DoGDot(double p) const {
...
+// @p point. The range is split into equal @p partitions. The search can be early terminated if any of the
+// extents of the range at each iteration is within @p tolerance.
+// @param curve The SpiralGoundCurve to use.
+// @param point The point in the 2D INERTIAL-Frame to find a p parameter value for.
+// @param p0 The p parameter start range value. It must be non-negative.
+// @param p1 The p parameter end range value. It must be greater than @p p0 by GroundCurve::kEpsilon.
+// @param partitions The number of partitions of the range [@p p0; @p p1]. It must be greater or equal to 2.
+// @param tolerance The tolereance to early stop the search. It must be non-negative.
+// @return The p parameter value that minimizes the distance against @p point.
+double FindBestPParameter(const SpiralGroundCurve& curve, const maliput::math::Vector2& point, double p0, double p1,
```
Comment (francocipollone) [inline src/maliput_malidrive/road_curve/spiral_ground_curve.cc:201]: Not sure how it performs this binary search here however using the `maliput::math::KDTree` is possible here. The kdtree should be loaded at construct time(which might require some time when loading the map as you need to evaluate G at all the points) and in each gIneverse call the `nearest_point` method which will be quick

**Review by francocipollone — APPROVED** [review]
LGTM!

**Comment by agalbachicar** [comment]
Declining in favor of a new implementation.

**Pull request closed without merging**